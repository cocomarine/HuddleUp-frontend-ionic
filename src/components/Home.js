import React from "react";
import { useNavigate } from "react-router-dom";
import { useAuthContext } from "../hooks/useAuthContext";

import "../styles/common/titles.css";
import "../styles/common/page.css";
import "../styles/common/buttons.css";
import "../styles/home.css";

const Home = () => {

  const { user } = useAuthContext();
  
  const navigate = useNavigate();

  const changeLocation = (redirect) => {
    navigate(redirect, { replace: true });
    // window.location.reload();
  };

  return (
    <>
      <div className="homeBody">
        <h1 className="home-title">
          <font color="#ED2643">h</font>
            uddle
          <font color="#2284B5">U</font>
            p
          </h1>
        {user ? <>
          <div className="homeText subtitle">
          Let's plan and get together!
          </div>
          <div className="btns logged-in">
            <button
              className="myEventsBtn link-button"
              onClick={() => {
                changeLocation("/myevents");
              }}
            >
              My Events
            </button>
            <button
              className="createEventBtn link-button"
              onClick={() => {
                changeLocation("/createevent");
              }}
            >
              Create Event
            </button>
            <button
              className="joinEventBtn link-button"
              onClick={() => {
                changeLocation("/joinevents");
              }}
            >
              Join Event
            </button>
          </div> 
        </> :<>
          <div className="homeText subtitle">
          Join us and plan your events and activities!
          </div>
          <div className="btns logged-out">
            <button
              className="loginBtn link-button"
              onClick={() => {
                changeLocation("/login");
              }}
            >
              Login
            </button>

            <button
              className="signUpBtn link-button"
              onClick={() => {
                changeLocation("/signup");
              }}
            >
              Sign up
            </button>
          </div>
        </>}
      </div>
    </>
  );
};

export default Home;