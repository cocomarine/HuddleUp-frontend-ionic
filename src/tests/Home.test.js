import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Home from "../components/Home";
import { AuthContextProvider } from "../contexts/AuthContext";

const validProps = {
  onClick: jest.fn(),
  onSubmit: jest.fn(),
};

beforeEach(() => {
  render(
    <BrowserRouter>
      <AuthContextProvider>
        <Home onClick={validProps.onClick} onSubmit={validProps.onSubmit} />
      </AuthContextProvider>
    </BrowserRouter>
  );
});

describe("Home", () => {
  const { asFragment } = render(
    <BrowserRouter>
      <AuthContextProvider>
        <Home />
      </AuthContextProvider>
    </BrowserRouter>
  );
  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders the button Log In correctly", () => {
    const button = screen.getAllByRole("button");

    const loginButton = screen.getByText(/Login/i);
    expect(loginButton).toBeInstanceOf(HTMLButtonElement);
  });

  it("redirects you upon logging in being successful", () => {
    const loginButton = screen.getByText(/sign up/i);
    fireEvent.click(loginButton);
  });
});
