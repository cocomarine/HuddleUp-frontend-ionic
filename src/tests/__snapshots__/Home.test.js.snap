// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<DocumentFragment>
  <div
    class="homeBody"
  >
    <h1
      class="home-title"
    >
      <font
        color="#ED2643"
      >
        h
      </font>
      uddle
      <font
        color="#2284B5"
      >
        U
      </font>
      p
    </h1>
    <div
      class="homeText subtitle"
    >
      Join us and plan your events and activities!
    </div>
    <div
      class="btns logged-out"
    >
      <button
        class="loginBtn link-button"
      >
        Login
      </button>
      <button
        class="signUpBtn link-button"
      >
        Sign up
      </button>
    </div>
  </div>
</DocumentFragment>
`;
